hub/test/src/testServer.ts

```diff
@@ -472,6 +474,136 @@ export function testServer() {
     })
   })
 
+  test('handle archival writes', async (t) => {
+    await usingMemoryDriver(async (mockDriver) => {
+      const scopes = [
+        {
+          scope: 'putFileArchival',
+          domain: '/foo/bar',
+        },
+        {
+          scope: 'putFileArchivalPrefix',
+          domain: 'baz'
+        }
+      ]
+
+      const server = new HubServer(mockDriver, new MockProofs(),
+                                  { whitelist: [testAddrs[0]], serverName: TEST_SERVER_NAME,
+                                    authTimestampCacheSize: TEST_AUTH_CACHE_SIZE })
+      const challengeText = auth.getChallengeText(TEST_SERVER_NAME)
+
+      const authPart = auth.V1Authentication.makeAuthPart(testPairs[0], challengeText, undefined, undefined, scopes)
+
+      const authorization = `bearer ${authPart}`
+      const authenticator = auth.parseAuthHeader(authorization)
+      t.throws(() => authenticator.isAuthenticationValid(testAddrs[1], [challengeText]),
+              errors.ValidationError, 'Wrong address must throw')
+      t.throws(() => authenticator.isAuthenticationValid(testAddrs[0], ['potatos are tasty']),
+              errors.ValidationError, 'Wrong challenge text must throw')
+      t.ok(authenticator.isAuthenticationValid(testAddrs[0], [challengeText]),
+          'Good signature must pass')
+
+      // scopes must be present
+      const authScopes = authenticator.getAuthenticationScopes()
+      t.equal(authScopes[0].scope, 'putFileArchival', 'scope 0 is putFileArchival')
+      t.equal(authScopes[0].domain, '/foo/bar', 'scope 0 is for /foo/bar')
+      t.equal(authScopes[1].scope, 'putFileArchivalPrefix', 'scope 1 is putFileArchivalPrefix')
+      t.equal(authScopes[1].domain, 'baz', 'scope 1 is for baz')
+
+      const getDataStream = () => {
+        const s = new PassThrough()
+        s.end('hello world')
+        return s
+      }
+
+      // test .history. file write rejection
+      try {
+        await server.handleRequest(testAddrs[0], 'baz/.history.not_ok.txt', { 
```

Comment:
I think this should be allowed -- we're going to prepend `.history.` ... to the file name anyway.

Reply:
Done